solutions/communication_kafka/README.md

```diff
@@ -0,0 +1,423 @@
+   - it's flexible, scalable and fault-tolerant 
+   - Data operations might be stateless, stateful or windowed.
+   - non-buffering system => streams are processed one at a time
+   
+
+2. Logging
+
+   Advantages
+   - supports many integrations
+   - a clearly readable data set
+   - simple Implementation
+   - provides organizations access to a large number of tools 
+
+   Disadvantages
+   - Log messages are maintained for a certain amount of time before   
+     deleting them
+   - During a spike, events can be deleted or lost without a user's 
+     knowledge while they are timed out
+   - lack of scalability within a large enterprise environment
+   - An expanding company will have to increase the number pf resources on 
+     servers, as well as process more logs.
+
+3. Message Broker 
+
+   Advantages
+   - Enables communication between several services that may not be   
+     executing at the same time.
+   - Asynchronous processing improves system performance.
+   - Ensuring that messages are delivered reliably
+  
+   Disadvantages
+   - Increasing the complexity of the system
+   - Debugging can be difficult
+   - Learning takes time
+
+
+4. Retry
+
+    Advantages:
+    - Real time traffic is not disrupted (Batch processing can be unblocked)
+    - Error handling is observable (metadata can be obtained on the retries)
+    - Higher real-time throughput can be achieved
+
+    Disadvantage:
+    - Kafka's ordering guarantees for the topic on which non-blocking retries is implemented is no longer applied.
+    - The batch processing is blocked, as failed messages are repeatedly processed in real time (they are re-consumed again and again).
+
+5. Data Monitoring (Metrics)
+    
+    Advantages:
+   Assist in identifying potential bottlenecks and performance concerns before they cause major issues
+
+    Disadvantages:
+    - Kafka lacks a full set of management and monitoring capabilities. This makes enterprise support staff wary about implementing kafka and maintaining it in the long run.
+    - The first challenge above can be overcome by installing third-party, open-source, or commercial graphical tools that offer administrative and monitoring features.
+
+6. Event Sourcing
+
+    Advantages:
+    - Kafka can be use as a natural backbone for Event Sourcing
+    Disadvantages:
+    
+    
+7. Tracking (Activity Tracking) 
+
+    Advantages: -
+    
+    
+    Disadvantages:
+    - often requires a very high volume of throughput 
+    
+    
+8. Temporary/Permanent Data Storing 
+
+    Advantages:
+    - can be easily integrated with databases and cloud data lake storage
+    - quite easy to understand
+    - helps to avoid copying the full dump of one database to another
+    
+    
+    Disadvantages:
+    - doesn’t provide arbitrary access lookup to the data
+    - there isn’t a query language like SQL available to help you access data
+    - it is not a replacement for databases
+
+# Context and Scope (in progress...)
```

Comment:
This section is quite important for you I guess. 

For me the scope is really limited to how can you implement
- logging
- monitoring 
- retry 
with spring-kafka